From 21312b2a8961e173780f76b0cb58f11b2e96f43d Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sat, 20 May 2023 18:44:19 +0200
Subject: [PATCH 816/985] evl: remove module pointer from calls to
 class_create()

Since #1aaba11da, class_create() stops taking a module pointer. Fix up
all callers accordingly.

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 drivers/evl/hectic.c | 1 -
 drivers/evl/latmus.c | 1 -
 kernel/evl/factory.c | 4 ++--
 3 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/evl/hectic.c b/drivers/evl/hectic.c
index 55dc1acdb..dbc3b9dc5 100644
--- a/drivers/evl/hectic.c
+++ b/drivers/evl/hectic.c
@@ -650,7 +650,6 @@ static int hectic_release(struct inode *inode, struct file *filp)
 
 static struct class hectic_class = {
 	.name = "hectic",
-	.owner = THIS_MODULE,
 };
 
 static const struct file_operations hectic_fops = {
diff --git a/drivers/evl/latmus.c b/drivers/evl/latmus.c
index 65c8cbe51..b8731fc09 100644
--- a/drivers/evl/latmus.c
+++ b/drivers/evl/latmus.c
@@ -1172,7 +1172,6 @@ static int latmus_release(struct inode *inode, struct file *filp)
 
 static struct class latmus_class = {
 	.name = "latmus",
-	.owner = THIS_MODULE,
 };
 
 static const struct file_operations latmus_fops = {
diff --git a/kernel/evl/factory.c b/kernel/evl/factory.c
index f07bd702c..9ed127f61 100644
--- a/kernel/evl/factory.c
+++ b/kernel/evl/factory.c
@@ -740,7 +740,7 @@ static int create_element_class(struct evl_factory *fac)
 	if (fac->minor_map == NULL)
 		return ret;
 
-	class = class_create(THIS_MODULE, fac->name);
+	class = class_create(fac->name);
 	if (IS_ERR(class)) {
 		ret = PTR_ERR(class);
 		goto cleanup_minor;
@@ -882,7 +882,7 @@ int __init evl_early_init_factories(void)
 {
 	int ret;
 
-	evl_class = class_create(THIS_MODULE, "evl");
+	evl_class = class_create("evl");
 	if (IS_ERR(evl_class))
 		return PTR_ERR(evl_class);
 
-- 
2.30.2

